polars/polars-arrow/src/array/fixed_size_list.rs

```diff
@@ -81,12 +81,68 @@ impl<'a> AnonymousBuilder<'a> {
 
     pub fn finish(self, inner_dtype: Option<&DataType>) -> Result<FixedSizeListArray> {
         let inner_dtype = inner_dtype.unwrap_or_else(|| self.arrays[0].data_type());
-        let values = concatenate::concatenate(&self.arrays)?;
-        let dtype = ListArray::<i64>::default_datatype(inner_dtype.clone());
-        Ok(FixedSizeListArray::new(
-            dtype,
-            values,
-            self.validity.map(|validity| validity.into()),
-        ))
+        let inner_size = self.inner_size;
+        let validity = self.validity.map(|validity| validity.into());
+        let values = expanding_concatenate(&self.arrays, &validity, inner_dtype, inner_size)?;
```

Comment:
I'm struggling with the `finish()` call, especially in relation to the `push_null` above. An Arrow `FixedSizeListArray` doesn't have offsets, so for a null object we need to still allocate `inner_size` in the array. But in `push_null` we don't know what the inner data type is, and in any case we can't push an owned array to `self.arrays` because that vec is holding `&'a dyn Array`. 

The way I tried to fix that is by having `push_null` _only push validity_. Then in `finish()` we need to support _expanding_ the input arrays during the concatenation, so that we can add null entries to the concatenated array. It doesn't look like arrow2 handles this natively.

I hit other issues in my implementation of `expanding_concatenate`, so that could use some feedback.